### ISSUE TITLE
Expose Metadata functionality through World API instead.

### ISSUE THREAD

**Issue by bryanedds** [bryanedds/Nu#522]
Right now. Metadata has become a singleton, which is an anti-pattern because it exposes mutable global state. Lets remove the globality of Metadata's MetadataPackages and make it a member of World and only accessible from the World API.

**Comment by bryanedds** [comment]
Incorrect issue -

okay wait, now i memba why Metadata became a singleton
now that i think about it, it wasn't an accident
Metadata became a singleton in Nu so that metadata could be accessed from Content functions and other functions that don't take the World as a parameter.
The thing about Metadata is that the data in it general IS intended to be immutable.
the only time that changes actually happen in it is when Assets are reloaded.
so it's immutable, but it's also not
so i made a 'practical' decision in that regard
basically, we'll just say it's effectively immutable because it's immutable when the player is actually playing the game
it's only mutable during development processes
can i get away with that
for the most part i think i can

Closing.
